Technical explanation for `disableTLSCertificateValidation` (#13024)
NuGet/Home: proposed/2023/InsecureConnectionsDisableCertificateValidation.md

### Technical explanation

<!-- Explain the proposal in sufficient detail with implementation details, interaction models, and clarification of corner cases. -->

> This proposal introduces a feature to manage the `HttpClientHandler` for conditionally disabling TLS certificate validation. By setting a boolean flag, `disableTLSCertificateValidation`, to true, the HttpClientHandler is configured to bypass the standard certificate checks via a custom `ServerCertificateCustomValidationCallback` that unconditionally returns `true`. This means that any SSL/TLS certificate, regardless of its source, expiration status, or domain name mismatch, will be accepted as valid, effectively disabling the usual security checks performed for HTTPS connections. **Note** This is done per source level.

Reviewer: If you add a new line after every sentence, you'll build a habit that will make Andy super happy :) and also makes future edits smaller as text is spread across multiple lines.

Other nit edits:

```suggestion
This proposal introduces a feature to manage the `HttpClientHandler` for conditionally disabling TLS certificate validation. By setting a boolean flag, `disableTLSCertificateValidation`, to true, the `HttpClientHandler` is configured to bypass the standard certificate checks via a custom `ServerCertificateCustomValidationCallback` that unconditionally returns `true`. This means that any SSL/TLS certificate, regardless of its source, expiration status, or domain name mismatch, will be accepted as valid, effectively disabling the usual security checks performed for HTTPS connections. 

> [!Note]
> `disableTLSCertificateValidation` applies at the package source level.
```
